### PR TITLE
ENH: `stats.qmc`: faster hypercube point comparison and scrambling doc

### PR DESCRIPTION
**Pull request by tupui**
Few minor things on QMC:

* Fix scrambling doc in Sobol'. It's not using Owen (NUS) but something from Matousek (LMS+shift). This came from a personal communication with Pierre L'Ecuyer.
* Faster (and clearer) point within hypercube check.
* Refactor usage of deprecated `assert_almost_equal` and `assert_array_almost_equal`.
* Uniformize a bit more the bibliography entries.

### CODE REVIEW

**Comment by tupui** [comment]
@mdhaber can you take a look? I moved some code over from the Lloyd PR here and took the opportunity to update very minor things.

File: scipy/stats/_qmc.py
Comment (mdhaber) [inline scipy/stats/_qmc.py:167]: Can't review without making suggestions, right?

```suggestion
        if (sample.max() > 1.) or (sample.min() < 0.):
```

Comment (tupui) [reply scipy/stats/_qmc.py:167]: Haha sure!

Indeed, better than having the not.

File: scipy/stats/_qmc.py
Comment (tupui) [inline scipy/stats/_qmc.py:297]:
```suggestion
    if (sample.max() > 1.) or (sample.min() < 0.):
```

File: scipy/stats/_qmc.py
Comment (tupui) [inline scipy/stats/_qmc.py:360]:
```suggestion
    if (sample.max() > 1.) or (sample.min() < 0.):
```

File: scipy/stats/_qmc.py
Comment (mdhaber) [inline scipy/stats/_qmc.py:797]: Before:
image

Now:
image

Good.

File: scipy/stats/_qmc.py
Comment (mdhaber) [reply scipy/stats/_qmc.py:167]: Shall I wait for this to merge or go for it?

**Review by mdhaber — APPROVED** [review]
Looks like an improvement to me. I'll go ahead and merge now if you like or wait for you to tweak the within-hypercube checks.

**Comment by tupui** [comment]
> Looks like an improvement to me. I'll go ahead and merge now if you like or wait for you to tweak the within-hypercube checks.

Thanks, I just pushed your suggestion so yes you can go ahead.

**Comment by mdhaber** [comment]
OK I'll merge when tests are done. Or if I notice, when tests pass on Meson. I don't imagine this will be platform-dependent.

**Comment by mdhaber** [comment]
Thanks @tupui!

**Pull request merged**
